src/Razor/test/Microsoft.AspNetCore.Razor.VSCode.FunctionalTest/tests/GoToDefinition.test.ts

```diff
@@ -0,0 +1,119 @@
+            'vscode.executeDefinitionProvider',
+            cshtmlDoc.uri,
+            new vscode.Position(0, 18));
+
+        assert.equal(definitions!.length, 0, 'Should have had no results');
+    });
+
+    test('Definition inside file works', async () => {
+        const imports = path.join(mvcWithComponentsRoot, 'Views', '_ViewImports.cshtml');
+        const importDoc = await vscode.workspace.openTextDocument(imports);
+        const importEditor = await vscode.window.showTextDocument(importDoc);
+        const firstLine = new vscode.Position(0, 0);
+        await importEditor.edit(edit => edit.insert(firstLine, '@functions{\n void Action()\n{\n}\n}\n'));
+        await importEditor.edit(edit => edit.insert(firstLine, '@{\nAction();\n}\n'));
+        const definitions = await vscode.commands.executeCommand<vscode.Location[]>(
+            'vscode.executeDefinitionProvider',
+            importDoc.uri,
+            new vscode.Position(1, 2));
+
+        assert.equal(definitions!.length, 1, 'Should have had exactly one result');
+        const definition = definitions![0];
+        assert.ok(definition.uri.path.endsWith('_ViewImports.cshtml'));
+        assert.equal(definition.range.start.line, 4);
+    });
+
+    test('Definition outside file works', async () => {
+        const firstLine = new vscode.Position(0, 0);
+        await editor.edit(edit => edit.insert(firstLine, '@{\nvar x = typeof(Program);\n}\n'));
+
+        const definitions = await vscode.commands.executeCommand<vscode.Location[]>(
+            'vscode.executeDefinitionProvider',
+            cshtmlDoc.uri,
+            new vscode.Position(1, 17));
+
+        assert.equal(definitions!.length, 1, 'Should have had exactly one result');
+        const definition = definitions![0];
+        assert.ok(definition.uri.path.endsWith('Program.cs'), `Expected def to point to "Program.cs", but it pointed to ${definition.uri.path}`);
+        assert.equal(definition.range.start.line, 3);
+    });
+
+    test('Definition of javascript works in cshtml', async () => {
+        const firstLine = new vscode.Position(0, 0);
+        await editor.edit(edit => edit.insert(firstLine, `<script>
+    var abc = 1;
+    abc.toString();
+</script>
+`));
+        const definitions = await vscode.commands.executeCommand<vscode.Location[]>(
+            'vscode.executeDefinitionProvider',
+            cshtmlDoc.uri,
+            new vscode.Position(2, 5));
+
+        assert.equal(definitions!.length, 1, 'Should have had exactly one result');
+        const definition = definitions![0];
+        assert.ok(definition.uri.path.endsWith('Index.cshtml'), `Expected 'Index.cshtml', but got ${definition.uri.path}`);
+        assert.equal(definition.range.start.line, 1);
+    });
+
+    test('Definition of javascript works in razor', async () => {
+        const firstLine = new vscode.Position(0, 0);
+        const razorPath = path.join(mvcWithComponentsRoot, 'Components', 'Counter.razor');
```

Comment:
nit: If you want you can just re-use the above constructed `editor` since that document is already guarunteed to be opened/focused.

Reply:
This test is specifically testings `.razor` files, if we used the `editor` for `Index.cshtml` then it would be identical to the test above it and should just be deleted.

Reply:
oh derp, my bad 👍